react-app/src/modules/account/account.js

```diff
@@ -0,0 +1,98 @@
+import scrypt from 'scrypt-async';
+import nacl from 'tweetnacl';
+import sha256 from 'fast-sha256';
+import blake from 'blakejs';
+
+nacl.util = require('tweetnacl-util');
+
+let _username, _pk, _sk, _tz, _offset;
+
+export function clearAccount() {
+  _username = _pk = _sk = _tz = _offset = undefined;
+}
+
+export function _setKeyPair(pk, sk) {
+  _pk = nacl.util.decodeBase64(pk);
+  _sk = nacl.util.decodeBase64(sk);
+}
+
+function _generateNonce(publicKeyA, publicKeyB) {
+  let state = blake.blake2bInit(nacl.box.nonceLength, null);
+  blake.blake2bUpdate(state, publicKeyA);
+  blake.blake2bUpdate(state, publicKeyB);
+  return blake.blake2bFinal(state);
+}
+
+export function constructor(username, password) {
+  return new Promise(resolve => {
+    scrypt(sha256(nacl.util.decodeUTF8(password)), sha256(nacl.util.decodeUTF8(username)), {
+      N: 1 << 20,
+      r: 8,
+      p: 1,
+      dkLen: 32,
+      encoding: 'binary'
+    }, function(derivedKey) {
+      const keyPair = nacl.box.keyPair.fromSecretKey(derivedKey);
+      _username = username;
+      _pk = keyPair.publicKey;
+      _sk = keyPair.secretKey;
+      resolve(null);
+    });
+  });
+}
+
+export function bytes() {
+  return JSON.stringify({
+    username: _username,
+    pk: _pk,
+    tz: _tz,
+    offset: _offset
+  });
+}
+
+export function encrypt(plainText) {
+  const messageToEncrypt = nacl.util.decodeUTF8(plainText);
+  const ephemeralKeyPair = nacl.box.keyPair();
+  const nonce = _generateNonce(ephemeralKeyPair.publicKey, _pk);
+  const boxed = nacl.box(messageToEncrypt, nonce, _pk, ephemeralKeyPair.secretKey);
+
+  let output = new Uint8Array(nacl.box.publicKeyLength + boxed.length);
+  output.set(ephemeralKeyPair.publicKey);
+  output.set(boxed, nacl.box.publicKeyLength);
+
+  return nacl.util.encodeBase64(output);
+}
+
+export function decrypt(cipherText) {
+  const input = nacl.util.decodeBase64(cipherText);
+  const originalEphemeralPublicKey = input.subarray(0, nacl.box.publicKeyLength);
+
+  const nonce = _generateNonce(originalEphemeralPublicKey, _pk);
+
+  const boxData = input.subarray(nacl.box.publicKeyLength);
+  return nacl.util.encodeUTF8(nacl.box.open(boxData, nonce, originalEphemeralPublicKey, _sk));
+}
+
+export function getUsername() {
```

Comment:
When the variable is named `_something`, I'm more of a fan of having the getter named `something()`, but maybe that's a more Go-ish kind of thing, @ivanalejandro0 / @leanbalma any thoughts on this kind of convention?

Reply:
According to what I read recently, JS objects' getters/setters use the prop name directly, so I think is more standard. I will improve this together with the implementation of account as a class.

Reply:
I've seen both getter/setter or `getSomething`, I think it's a matter of conventions.
I haven't seen mane `something()` for `_something`, though. I think that's usually a syntax for a function that does a specific task, not that returns a value.
But I guess nothing's forbidden on the JS realm, right?